src/iter/mod.rs

```diff
@@ -233,11 +233,15 @@ pub trait ParallelIterator: Sized {
     fn reduce_with<OP>(self, op: OP) -> Option<Self::Item>
         where OP: Fn(Self::Item, Self::Item) -> Self::Item + Sync
     {
-        self.map(Some).reduce(|| None, |opt_a, opt_b| match (opt_a, opt_b) {
-            (Some(a), Some(b)) => Some(op(a, b)),
-            (Some(v), None) | (None, Some(v)) => Some(v),
-            (None, None) => None,
-        })
+        self.fold(|| None, |opt_a, b| match opt_a {
```

Comment:
I see. That's true. I don't have a strong opinion, just curious. I agree that it seems like this formulation *might* unlock some optimization somehow.